src/wallet/wallet.h

```diff
@@ -1096,11 +1166,38 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
     bool LoadCryptedSaplingZKey(const libzcash::SaplingExtendedFullViewingKey &extfvk,
                                 const std::vector<unsigned char> &vchCryptedSecret);
 
-    /**
-     * Unified keys & addresses
-     */
-    libzcash::ZcashdUnifiedSpendingKey GenerateNewUnifiedSpendingKey();
-    std::optional<libzcash::ZcashdUnifiedSpendingKey> GenerateUnifiedSpendingKeyForAccount(libzcash::AccountId accountId);
+    //
+    // Unified keys & addresses
+    //
+
+    //! Generate the unified spending key from the wallet's mnemonic seed
+    //! for the next unused account identifier.
+    std::pair<libzcash::ZcashdUnifiedSpendingKey, ZcashdUnifiedSpendingKeyMetadata>
+        GenerateNewUnifiedSpendingKey();
+
+    //! Generate the next available unified spending key from the wallet's
+    //! mnemonic seed.
+    std::optional<std::pair<libzcash::ZcashdUnifiedSpendingKey, ZcashdUnifiedSpendingKeyMetadata>>
+        GenerateUnifiedSpendingKeyForAccount(libzcash::AccountId accountId);
+
+    //! Retrieves the UFVK derived from the wallet's mnemonic seed for the specified account.
+    std::optional<std::pair<libzcash::UFVKId, libzcash::ZcashdUnifiedFullViewingKey>>
```

Comment:
This returns the `ZcashdUnifiedFullViewingKey`, which may represent only a subset of the UFVK from which it was derived, so we need that `UFVKId`. I had at one point made `UFVKId` a member of `ZcashdUnifiedFullViewingKey` for this purpose, but that gets complicated because then it requires the `KeyConstants` to be passed in to all the static constructor functions (in order to properly be able to generate the UFVKId for a newly created UFVK, since the UFVKId depends on the serialized form.) 

I guess there's another way to approach this that I hadn't considered before, which is to make the derivation of `ZUSK -> ZUFVK` instead be `ZUSK -> UFVK` and leave it up to the caller to be able to perform the `(KeyConstants, UFVK) -> ZUFVK` operation. I'll give that a try and see whether it works.